modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -174,5 +188,23 @@ def enum_zones
       logger.debug "Enumerated authoritative dns zones: #{zones}"
       zones
     end
+
+    def enum_records(zone_name, node_name, type)
+      records = []
+      response = execute "/EnumRecords #{zone_name} #{node_name}. /Type #{type}", "EnumRecords", true
+      response.each do |line|
+        line.chomp!
+        logger.debug "Extracting record from dnscmd output '#{line}'"
+        matches = /^@\s+(?<ttl>\d+)\s+(?<type>#{type})\s+(?<record>\S+)$/.match(line)
```

Comment:
I don't *need* to capture `ttl` or `type`, but I thought it made the regex nicely self documenting.  My first iteration actually had the regex over several lines with comments ;)

It should be ok to use `[[:upper:]]` for matching the type, but `#{type}` is available and tighter?

I'm happy to change it though!

I'm still a ruby novice.  Could you give a more complete example of how I'd use `=~`?
```
if line =~ /^@\s+(?<ttl>\d+)\s+(?<type>#{type})\s+(?<record>\S+)$/
  #Found a match, but where do the captures go?
else
  #Not a matching line
end
```